src/main/scala/scorex/util/serialization/VLQWriter.scala

```diff
@@ -0,0 +1,169 @@
+    while (true) {
+      if ((value & ~0x7FL) == 0) {
+        buffer(position) = value.asInstanceOf[Byte]
+        position += 1
+        putBytes(Arrays.copyOf(buffer, position))
+        return this
+      } else {
+        buffer(position) = ((value.asInstanceOf[Int] & 0x7F) | 0x80).toByte
+        position += 1
+        value >>>= 7
+      }
+    }
+    this
+    // see https://rosettacode.org/wiki/Variable-length_quantity for implementations in other languages
+  }
+
+  @inline override def putBits(xs: Array[Boolean]): this.type = {
+    if (xs.isEmpty) return this
+    val bitSet = new BitSet(xs.length)
+    xs.zipWithIndex.foreach { case (bool, i) => bitSet.set(i, bool)}
+    // pad the byte array to fix the "no bit was set" behaviour
+    // see https://stackoverflow.com/questions/11209600/how-do-i-convert-a-bitset-initialized-with-false-in-a-byte-containing-0-in-java
+    val bytes = Arrays.copyOf(bitSet.toByteArray, (xs.length + 7) / 8)
+    putBytes(bytes)
+    this
+  }
+
+  @inline override def putOption[T](x: Option[T])(putValue: (this.type, T) => Unit): this.type = {
+    x match {
+      case Some(v) =>
+        put(1.toByte)
+        putValue(this, v)
+      case None =>
+        put(0.toByte)
+    }
+    this
+  }
+
+  /**
+    * Encode String is shorter than 256 bytes
+    *
+    * @param s String
+    * @return
+    */
+  override def putShortString(s: String): this.type = {
+    val bytes = s.getBytes
+    require(bytes.size < 256)
+    putUByte(bytes.size.toByte)
+    putBytes(bytes)
+    this
+  }
+}
+
+object VLQWriter {
+
+  /**
+    * Encode a ZigZag-encoded 32-bit value.  ZigZag encodes signed integers
+    * into values that can be efficiently encoded with varint.  (Otherwise,
+    * negative values must be sign-extended to 64 bits to be varint encoded,
+    * thus always taking 10 bytes on the wire.)
+    *
+    * @see [[https://developers.google.com/protocol-buffers/docs/encoding#types]]
+    *
+    * @param n signed Int
+    * @return unsigned Int stored in a signed Int
+    */
+  def encodeZigZagInt(n: Int): Int =
+  // Note:  the right-shift must be arithmetic
+  // source: http://github.com/google/protobuf/blob/a7252bf42df8f0841cf3a0c85fdbf1a5172adecb/java/core/src/main/java/com/google/protobuf/CodedOutputStream.java#L934
+    (n << 1) ^ (n >> 31)
+
+  /**
+    * Encode a ZigZag-encoded 64-bit value.  ZigZag encodes signed integers
+    * into values that can be efficiently encoded with varint.  (Otherwise,
+    * negative values must be sign-extended to 64 bits to be varint encoded,
+    * thus always taking 10 bytes on the wire.)
+    *
+    * @see [[https://developers.google.com/protocol-buffers/docs/encoding#types]]
+    * @param n signed Long
+    * @return unsigned Long stored in a signed Long
+    */
+  def encodeZigZagLong(n: Long): Long =
```

Comment:
@oskin1 already commented in a different place to add {} braces.
Also why the comment here is separated from method comments?